### PR TITLE
Bump Jedis version to 5.0.2

### PR DESCRIPTION
**Pull request by sekikn**
### Description

Currently, the redis-cache extension uses Jedis 2.9.0, which was released over seven years ago and [is no longer listed in the official support matrix](https://github.com/redis/jedis#supported-redis-versions). This patch upgrades it to ensure the compatibility with the recent version of Redis and make future upgrades easier, including:

* Upgrade Jedis to v5.0.2, the latest version at this writing, and address the API changes and dependency version mismatch.

* Replace mock-jedis with jedis-mock, since the former has not been actively maintained any longer and not compatible with recent versions of Jedis.

This PR has:

- [x] been self-reviewed.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

File: extensions-contrib/redis-cache/src/main/java/org/apache/druid/client/cache/RedisClusterCache.java [hunk 1]
```diff
@@ -110,10 +109,6 @@ protected Pair<Integer, Map<NamedKey, byte[]>> mgetFromRedis(Iterable<NamedKey>
   @Override
   protected void cleanup()
   {
-    try {
-      cluster.close();
-    }
-    catch (IOException ignored) {
-    }
+    cluster.close();
```
Comment (abhishekagarwal87) [inline extensions-contrib/redis-cache/src/main/java/org/apache/druid/client/cache/RedisClusterCache.java:112]: why this change?

**Review by abhishekagarwal87 — APPROVED** [review]

**Comment by abhishekagarwal87** [comment]
LGTM. Thank you for your contribution @sekikn

File: extensions-contrib/redis-cache/src/main/java/org/apache/druid/client/cache/RedisClusterCache.java (hunk 1 above)
Comment (sekikn) [reply extensions-contrib/redis-cache/src/main/java/org/apache/druid/client/cache/RedisClusterCache.java:112]: Thank you for the review @abhishekagarwal87! In Jedis 5.0.2, that method is inherited from the `UnifiedJedis` class and [it doesn't raise IOException since `IOUtils.closeQuietly` is internally used](https://github.com/redis/jedis/blob/v5.0.2/src/main/java/redis/clients/jedis/UnifiedJedis.java#L233-L236).
